controller_manager/test/test_controllers_chaining_with_controller_manager.cpp

```diff
@@ -610,6 +679,271 @@ TEST_P(
+    pid_right_wheel_controller, PID_RIGHT_WHEEL,
+    test_chainable_controller::TEST_CONTROLLER_CLASS_NAME);
+
+  CheckIfControllersAreAddedCorrectly();
+
+  ConfigureAndCheckControllers();
+
+  // Set ControllerManager into Debug-Mode output to have detailed output on updating controllers
+  cm_->get_logger().set_level(rclcpp::Logger::Level::Debug);
+  rclcpp::get_logger("ControllerManager::utils").set_level(rclcpp::Logger::Level::Debug);
+
+  // at beginning controllers are not in chained mode
+  EXPECT_FALSE(pid_left_wheel_controller->is_in_chained_mode());
+  EXPECT_FALSE(pid_right_wheel_controller->is_in_chained_mode());
+  ASSERT_FALSE(diff_drive_controller->is_in_chained_mode());
+
+  // Test Case 1: Trying to activate a preceding controller when following controller is not activated
+  // --> return error (If STRICT); Preceding controller is still inactive.
+
+  // There is different error and timeout behavior depending on strictness
+  auto const getExpectedBehaviorVals = [&]
+  {
+    if (
+      test_param.strictness ==
+      controller_manager_msgs::srv::SwitchController::Request::STRICT)  // STRICT
+    {
+      return std::make_tuple(
+        controller_interface::return_type::ERROR, std::future_status::ready,
+        lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE);
+    }
+    else if (
+      test_param.strictness ==
+      controller_manager_msgs::srv::SwitchController::Request::BEST_EFFORT)  // BEST EFFORT
+    {
+      return std::make_tuple(
+        controller_interface::return_type::OK, std::future_status::timeout,
+        lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE);
+    }
+    else
+    {
+      std::runtime_error("test_param was not defined!");
+    }
+  };
+  auto const expected = getExpectedBehaviorVals();
+
+  // Attempt to activate preceding controller (diff-drive controller) with no check
+  ActivateController(DIFF_DRIVE_CONTROLLER, std::get<0>(expected), std::future_status::ready);
+
+  // Check if the controller activated (Should not be activated)
+  ASSERT_EQ(
+    lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, diff_drive_controller->get_state().id());
+
+  // Test Case 2: Try to activate a preceding controller the same time when trying to deactivate a following controller (using switch_controller function)
+  // --> return error; preceding controller is not activated, BUT following controller IS deactivated
+
+  // Activate and check the following controllers (pid_left_wheel_controller) (pid_right_wheel_controller)
+  ActivateAndCheckController(
+    pid_left_wheel_controller, PID_LEFT_WHEEL, PID_LEFT_WHEEL_CLAIMED_INTERFACES, 1u);
+  ActivateAndCheckController(
+    pid_right_wheel_controller, PID_RIGHT_WHEEL, PID_RIGHT_WHEEL_CLAIMED_INTERFACES, 1u);
+
+  // Attempt to activate a preceding controller (diff-drive controller) while trying to deactivate a following controller
+  switch_test_controllers(
+    {DIFF_DRIVE_CONTROLLER}, {PID_LEFT_WHEEL, PID_RIGHT_WHEEL}, test_param.strictness,
```

Comment:
Let's try to deactivate only one. The behavior should be the same.
```suggestion
    {DIFF_DRIVE_CONTROLLER}, {PID_RIGHT_WHEEL}, test_param.strictness,
```